observatory-dags/observatory/dags/telescopes/oapen_metadata.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2020 Curtin University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Author: Aniek Roelofs
+
+import csv
+import logging
+import os
+from datetime import datetime
+
+import jsonlines
+import pendulum
+from airflow.exceptions import AirflowException
+from airflow.models.taskinstance import TaskInstance
+from observatory.platform.telescopes.stream_telescope import (StreamRelease, StreamTelescope)
+from observatory.platform.utils.airflow_utils import AirflowVars
+from observatory.platform.utils.telescope_utils import convert
+from observatory.platform.utils.template_utils import upload_files_from_list
+from observatory.platform.utils.url_utils import get_ao_user_agent, retry_session
+
+
+# TODO
+# isbn have '-' sometimes, so not an integer, leave in?
+
+class OapenMetadataRelease(StreamRelease):
+    def __init__(self, dag_id: str, start_date: pendulum.Pendulum, end_date: pendulum.Pendulum, first_release: bool):
+        super().__init__(dag_id, start_date, end_date, first_release)
+
+    @property
+    def csv_path(self) -> str:
+        """ Path to store the original oapen metadata csv file"""
+        return os.path.join(self.download_folder, 'oapen_metadata.csv')
+
+    @property
+    def transform_path(self) -> str:
+        """ Path to store the transformed oapen metadata file"""
+        return os.path.join(self.transform_folder, 'metadata.jsonl')
```

Comment:
Would the files be able to be gzipped? They could be if under 4GB each: https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-json#limitations

I gzipped the GRID json files, the code here could be turned into a utility function to save as .jsonl.gz: https://github.com/The-Academic-Observatory/observatory-platform/blob/feature_telescope_template/observatory-dags/observatory/dags/telescopes/grid.py#L129-L137

Reply:
I think that's a good idea. The file is around ~50mb, so definitely under 4GB.